RocketChat/Rocket.Chat
# [EE] Improve Forwarding Department behaviour with Waiting queue feature

## Proposed changes (including videos or screenshots)





## Issue(s)

**Old Behaviour:**
Currently when we forward a chat to a particular department and all the agents in target department are at maximum capacity, the chat either not getting transferred (on or after v3.13.0) or the user were getting an error saying **no-agents are currently online**(before v3.13.0).

**New Behaviour:**
In the new behaviour we're proposing, if a chat is forwarded and all agents in the target department are at maximum capacity, the chat will get transferred to the target department's waiting queue. This means that the chat will be removed from the current serving agent's chats and then will be moved to the target department's waiting queue. The visitor will see their current position within the queue.

Note: This feature will not affect the `Manual Selection` Omnichannel Routing method in any way.
 
## Steps to test or reproduce

1. Goto Admin -> Omnichannel -> Routing
a. Set `Omnichannel Routing Method` to either `Auto Selection` or `Load Balancing`. 
b. Turn on `Waiting Queue`.
c. Set `Max. number of simultaneous chats` to `1` (one)
2. Create 2 departments, for and add one agents to each one of them respectively eg: **TestDept-1** and **TestDept-2** with **Agent-1** and **Agent-2** respectively.
3. Start 2 Visitor conversation with each of the Department. After both the chats is started, forward the chat from one department to another. eg: Transfer from **Agent-1** from **TestDept-1** to **TestDept-2**. 
4. **OLD BEHAVIOUR:** If you're on version **v3.13.0 or above**, you should see that the chat is not transferred. If you're below **v1.13.0**, you should see an error message saying something like, **no agents available**.
5. **NEW BEHAVIOUR:** The chat should be transferred to the target department's waiting queue and will be removed from the old department. In our example above: Agent-1 will be removed from **TestDept-1** and will be put to **Test-Dept-2**. Its queue position will be **1**. Once **Agent-2** closes the existing chat, the forwarded chat should assigned to the **Agent-2**.


## Further comments
